src/BattleServer/battle.cpp

```diff
@@ -1826,8 +1826,6 @@ void BattleSituation::useAttack(int player, int move, bool specialOccurence, boo
             calculateTypeModStab();
 
             calleffects(player, target, "BeforeCalculatingDamage");
-            /* For charge */
-            callpeffects(player, target, "BeforeCalculatingDamage");
```

Comment:
I opted to remove this because Line 1888 it clears the boost for Gen 5+ (Gen 3 and 4 are TurnMemory, Gen 5+ is ChainBP)